server/src/main/java/io/crate/execution/dml/upsert/TransportShardUpsertAction.java

```diff
@@ -290,6 +264,25 @@ protected WritePrimaryResult<UpsertReplicaRequest, ShardResponse> processRequest
                 break;
             }
         }
+        List<ColumnIdent> insertColumnNamesForReplica = insertColumnsForReplica.stream().map(Reference::column).toList();
+        List<Reference> insertOnConflictColumns = indexer.indexOrder().stream().map(ref -> {
+            int idx = insertColumnNamesForReplica.indexOf(ref.column());
+            if (idx != -1 && !insertColumnsForReplica.get(idx).valueType().equals(ref.valueType())) {
+                return insertColumnsForReplica.get(idx);
+            }
+            return ref;
+        }).toList();
```

Comment:
What's going on here? Why is it reretrieving the same column with a valueType check?

Reply:
Please ignore this, it is removed.